CONTRIBUTING.md

```diff
@@ -0,0 +1,120 @@
+# Contributing to Merlin
+
+Merlin is a community oriented open-source project and we encourage and value
+any kind of contribution. Thanks for taking the time to contribute 🐫 !
+
+## Code of Conduct
+
+Merlin  adheres to the OCaml Code of Conduct as stated in the [Code of Conduct
+document](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this
+code. Please report unacceptable behavior either to local contacts (listed in
+[here](CODE_OF_CONDUCT.md)) or to someone listed in the upstream [OCaml Code of
+Conduct](CODE_OF_CONDUCT.md).
+
+## Documentation
+
+Merlin (partial and fragmented) documentation can be found in the following
+places:
+- Installation instructions and editor setup can be found in the
+  [README.md](README.md) and the [Github Pages](https://ocaml.github.io/merlin/)
+- Various topics related to editor support and configuration can be found in the
+  [wiki](https://github.com/ocaml/merlin/wiki).
+- The Merlin protocol is described in [this
+  document](https://github.com/ocaml/merlin/blob/master/doc/dev/PROTOCOL.md).
+
+Contribution to the documentation are welcome!
+
+## Question, bug reports and feature requests
+
+We rely on [Github's issue tracker](https://github.com/ocaml/merlin/issues) for
+support questions, feature requests and bug reports.
+
+When reporting an issue, please include a precise reproduction in the bug report
+when that's possible, as it is a very useful tool to investigate. You should
+also check that you are using the latest version of Merlin and that a similar
+issue as not already been submitted.
+
+## Code contributions
+
+ ### Legal Notice
+ When contributing to Merlin, you must agree that you have authored 100% of the
+ content, that you have the necessary rights to the content and that the content
+ you contribute may be provided under the [project's MIT
+ license](https://github.com/ocaml/merlin/blob/master/LICENSE).
+
```

Comment:
(Disclaimer: I am *not* a lawyer)

Was this proposed Legal Notice framed by you or is it something that is being used by other projects also ?

Many contributions to ocaml merlin may modify and reuse existing ocaml merlin code, for example, so I wonder if the the stipulation of 100% authorship requirement in the proposed Legal Notice will be correct here.

Have you considered using something using the Developer Certificate of Origin 1.1 ? It is used by a lot of open source projects and may fulfill the objectives you had when you put up the proposed Legal Notice.

Reply:
I think CLAs, in general, apply to more ambiguous license cases. In the first instance, I think the request, albeit vague (for example, how to apply it to version migrations, which vendor existing code), to be "morally the author of a patch" is sufficient.

Or maybe it's better just to delete this paragraph?

Reply:
This proposed "Legal Notice" seems to try to obtain some protections given by typical CLAs. However, my main objection is that the language of the "Legal Notice" feels informal / may not cover all contribution scenarios where, say, for example, some of the contribution may be a modified / improved / built-upon version of code that already exists in merlin. In that case 100% authorship required by the proposed "Legal Notice" will probably not be met by that contribution.

> "morally the author of a patch" is sufficient

Yes, agree, but whatever language is used needs to be accurate and precise, because after all, it has the heading "Legal Notice".

In conclusion, if the merlin project needs explicity outlined protections from contributions , it should use something more formal and accurate. I have suggested the Developer Certificate of Origin 1.1 as a possible solution; there are of course more ways to achieve this via proper CLAs etc.

In my opinion, more thought must put into this proposed "Legal Notice".

Reply:
@sidkshatriya For the time being, what do you think about dropping the "Legal notice" section?

Reply:
> @sidkshatriya For the time being, what do you think about dropping the "Legal notice" section?

I've already expressed my personal opinion above on this proposed "Legal Notice" -- I find it problematic for the reason's outined above. @voodoos has not replied to these comments so I'm not aware of their opinion on the points we have been discussing. I had suggested an alternative (Developer Certificate of Origin) too but I lack context on the motivation for this proposed change.

OCaml merlin project owners/managers should make a decision on any new requirements/stipulations for future contributions and if they want to add the proposed "Legal Notice" or not. (I am of course not part of any decision making structure on this project).

Again as a personal opinion, contributing to open source projects should be as simple and straightforward as possible. If @xvw 's suggestion on dropping the proposed "Legal Notice" keeps things simple then the project owners/managers should consider it.

Reply:
Thanks for having a look and raising our awareness @sidkshatriya ! I agree that the language is blurry and it probably, at best, achieve nothing in the current state. There is no deep thinking or careful writing behind the current version, it was just a generic version found in a model that I used for drafting the document. I will remove it for now until more discussions happen.